packages/option/src/index.ts

```diff
@@ -0,0 +1,181 @@
+import { createSignal } from "solid-js";
+
+/**
+ * a utility class representing an Optional value, which can be either Some\<T> or None.
+ *
+ * @example
+ * ```ts
+ * const option = Some(5);
+ *
+ * if (option.isSome()) {
+ *   doSomething(option.unwrap())
+ * }
+ * ```
+ */
+class Option<T> {
+  #value: T | null;
+  constructor(value: T | null) {
+    this.#value = value;
+  }
+  /**
+   * Checks whether the option is the None variant.
+   * @example
+   * ```ts
+   * const userInput = None();
+   *
+   * if (userInput.isNone()) return alert("Invalid input");
+   * ```
+   */
+  isNone() {
+    return this.#value == null;
+  }
+  /**
+   * Checks whether the option is the Some variant.
+   * @example
+   * ```ts
+   * const userSubmission = Some(5);
+   *
+   * if (userSubmission.isSome()) return alert(`The user has already submitted value of ${userInput}`);
+   * ```
+   */
+  isSome() {
+    return this.#value != null;
+  }
+  /**
+   * Unwraps the stored value, possibly null.
+   * @example
+   * ```ts
+   * const count = Some(5);
+   *
+   * // safe
+   * onMount(() => {
+   *   doSomething(count.unwrap());
+   * });
+   *
+   * // very unsafe, possibly null!!!
+   * const onClick = () => {
+   *   doSomething(count.unwrap());
+   * };
+   *
+   * ```
+   */
+  unwrap() {
+    return this.#value;
+  }
+  /**
+   * Unwraps the stored value.
+   * @return the stored `Some` value or a provided default.
+   * @example
+   * ```ts
+   * const count = None();
+   *
+   * const onClick () => {
+   *   doSomething(count.unwrap_or(5));
+   * };
+   * ```
+   */
+  unwrap_or(value: T) {
+    return this.isNone() ? value : (this.#value as T);
+  }
+  /**
+   * Unwraps the stored value.
+   * @return the stored `Some` value or computes it from a closure.
+   * @example
+   * ```ts
+   * const count = None();
+   *
+   * const onClick () => {
+   *   doSomething(count.unwrap_or_else(() => 5));
+   * };
+   * ```
+   */
+  unwrap_or_else(closure: () => T) {
```

Comment:
Not sure how I feel about the snake case. It's obviously easier to read, but camelCase is the standard for js, especially for public APIs in libraries. So `unwrapOrElse` would be the way I think.